### ISSUE TITLE
[Bug] NCCL timeout when cfg_rate > 0.1

### ISSUE THREAD

**Issue by yjhong89** [hao-ai-lab/FastVideo#250]
### Environment
- Error occurs in both environments, H100 x 8 (set SP=4)
- PYTORCH 2.5.1, CUDA 12.4
- PYTORCH 2.4.1, CUDA 12.1

   

### Debug
- When cfg_rate > 0.1, the shape of prompt embed becomes different
  - https://github.com/hao-ai-lab/FastVideo/blob/d869d90d12746a53adf56330f34ffda77c38c390/fastvideo/dataset/latent_datasets.py#L47 (256, 4096)
  - https://github.com/hao-ai-lab/FastVideo/blob/d869d90d12746a53adf56330f34ffda77c38c390/fastvideo/dataset/latent_datasets.py#L50 (247, 4096)
  - But, I think it shouldn't be a problem if `train_sp_batch_size==1` but need to clarify why these shapes are different. (246 vs 247)
- Since I use 8 GPU and sp==4, rank 0/1/2/3 maps to nccl group 0, rank 4/5/6/7 maps to nccl group 1. 
  - When I print out logics in `train_one_step`, every batch elements from nccl group are fetched properly https://github.com/hao-ai-lab/FastVideo/blob/d869d90d12746a53adf56330f34ffda77c38c390/fastvideo/train.py#L111 
  - But, only one nccl group (such as 4/5/6/7) passes this function https://github.com/hao-ai-lab/FastVideo/blob/d869d90d12746a53adf56330f34ffda77c38c390/fastvideo/train.py#L112 and tensors from other nccl group hanged.

- I think this NCCL  error occurs when different shapes (or dtypes) tensors are loaded across nccl groups.

### Error
- When set `cfg_rate == 0.1`, https://github.com/hao-ai-lab/FastVideo/issues/228#issuecomment-2703249902 timeout error occurs
  - https://github.com/hao-ai-lab/FastVideo/blob/d869d90d12746a53adf56330f34ffda77c38c390/scripts/finetune/finetune_hunyuan.sh#L29

**Comment by BrianChen1129** [comment]
We also encountered this issue yesterday, and the reason is we hardcoded uncond_prompt and its mask with all 0 [here](https://github.com/hao-ai-lab/FastVideo/blob/d869d90d12746a53adf56330f34ffda77c38c390/fastvideo/dataset/latent_datasets.py#L33C14-L33C33). If you want to use cfg>0.0 to train the model, you might need to use text_encoder to encode unconditional prompt first. We will update this in the next version.

**Comment by yjhong89** [comment]
Okay, 
Then you mean that encode 1) prompt and 2) unconditional prompt both using this script right ?? 
- https://github.com/hao-ai-lab/FastVideo/blob/main/fastvideo/data_preprocess/preprocess_text_embeddings.py

And then load both embedding in here
- https://github.com/hao-ai-lab/FastVideo/blob/d869d90d12746a53adf56330f34ffda77c38c390/fastvideo/dataset/latent_datasets.py#L47

**Comment by yinian-lw** [comment]
change 256 to 257 sovled.
https://github.com/hao-ai-lab/FastVideo/blob/d869d90d12746a53adf56330f34ffda77c38c390/fastvideo/dataset/latent_datasets.py#L31
